files/task_helper.rb

```diff
@@ -46,7 +48,10 @@ def self.run
     # the task. Unhandled errors are caught and turned into an error result.
     # @param [Hash] params A hash of params for the task
     # @return [Hash] The result of the task
-    result = new.task(params)
+
+    tsk = new
+    tsk.context = RemoteContext.new(params) if params.key? :_target
```

Comment:
why not move this logic to the initialize method?

Reply:
Because `RemoteContext` need the params, which are read by this function
https://github.com/puppetlabs/puppetlabs-ruby_task_helper/blob/6c5b8c51f6f233a78587d9f80313057a31a85be1/files/task_helper.rb#L43-L46

This looked like the least ugly way of doing that